TFEngine.py

```diff
@@ -1746,25 +1753,37 @@ def search_single_seq(self, source, output_layer_name=None):
     dataset.init_seq_order(epoch=1)
     return self.search_single(dataset=dataset, seq_idx=0, output_layer_name=output_layer_name)
 
-  def search_single_bpe_to_bpe_seq(self, source, source_bpe, target_bpe, output_layer_name=None):
+  def search_single_string_to_string_seq(self, source, output_layer_name=None):
     """
     :param str source: source as a string
-    :param GeneratingDataset.BytePairEncoding source_bpe:
-    :param GeneratingDataset.BytePairEncoding target_bpe:
+    :param GeneratingDataset.Vocabulary source_voc:
+    :param GeneratingDataset.Vocabulary target_voc:
     :param str|None output_layer_name: e.g. "output". if not set, will read from config "search_output_layer"
     :return: list of all hyps, which is a tuple of score and string
     :rtype: list[(float,str)]
     """
-    assert source_bpe.num_labels == self.network.extern_data.data["data"].dim
-    assert target_bpe.num_labels == self.network.extern_data.data["classes"].dim
-    source_seq_list = source_bpe.get_seq(source)
+    assert self.source_voc != None
+    assert self.target_voc != None
+    assert self.source_voc.num_labels == self.network.extern_data.data["data"].dim
+    assert self.target_voc.num_labels == self.network.extern_data.data["classes"].dim
+    source_seq_list = self.source_voc.get_seq(source.split())
     results_raw = self.search_single_seq(source=source_seq_list, output_layer_name=output_layer_name)
     results = []
     for (score, raw) in results_raw:
-      txt = " ".join(map(target_bpe.labels.__getitem__, raw))
+      txt = self.target_voc.get_seq_labels(raw)
       results += [(score, txt)]
     return results
 
+  def translate(self, source): 
+    """
+    :param str source: source as a string. 
+    :return: tuple of (best score, best hyp)
+    :rtype: (float,str)
+    """   
+    self._checked_uninitialized_vars = True
+    results = self.search_single_string_to_string_seq(source)
+    return results[0]
```

Comment:
This function should be removed.
* It does not add really anything over `search_single_string_to_string_seq`.
* The naming bad. `Engine` works with any kind of task, on any kind of data. This is nothing MT related.
* `self._checked_uninitialized_vars = True` could be moved to `search_single_string_to_string_seq`.

Reply:
Ok, I agree here, will change.